view/next-project/src/components/sponsoractivities/SponsorActivitiesAddModal.tsx

```diff
@@ -58,26 +48,57 @@ export default function SponsorActivitiesAddModal(props: Props) {
     userID: users[0].id || 0,
     isDone: false,
     feature: 'なし',
+    design: 0,
+    url: '',
     expense: 0,
-    remark: '',
+    remark: REMARK_PAMPHLET,
   });
 
-  const [design, setDesign] = useState<string>('学生が作成');
-  useEffect(() => {
-    const newRemark =
-      design === '学生が作成'
-        ? REMARK_DESIGN_STUDENT + REMARK_PAMPHLET
-        : design === '企業が作成'
-        ? REMARK_DESIGN_SPONSOR + REMARK_PAMPHLET_SPONSOR
-        : REMARK_DESIGN_OTHER + REMARK_PAMPHLET_OTHER;
-    const newRemarkFeature =
-      formData.feature === 'ポスター'
-        ? REMARK_POSTER
-        : formData.feature === 'クーポン'
-        ? REMARK_COUPON
-        : '';
-    setFormData({ ...formData, remark: newRemark + newRemarkFeature });
-  }, [design]);
+  const setDesign =
+    (input: string) =>
+    (
+      e:
+        | React.ChangeEvent<HTMLSelectElement>
+        | React.ChangeEvent<HTMLInputElement>
+        | React.ChangeEvent<HTMLTextAreaElement>,
+    ) => {
+      const remarkOption =
+        formData.feature === 'ポスター'
+          ? REMARK_POSTER
+          : formData.feature === 'クーポン'
+          ? REMARK_COUPON
+          : '';
+      const newRemarkDesign = e.target.value === '0' ? REMARK_PAMPHLET : '';
+      console.log(formData.remark);
+      console.log(e.target.value);
+      setFormData({
+        ...formData,
+        design: Number(e.target.value),
+        remark: remarkOption + newRemarkDesign,
+      });
+    };
```

Comment:
不要なconsole.logは消した方がいいかも
あとsetFeatureと同様にinputは受け取らなくて大丈夫
```suggestion
  const setDesign = (e: React.ChangeEvent<HTMLInputElement>) => {
    const remarkOption =
      formData.feature === 'ポスター'
        ? REMARK_POSTER
        : formData.feature === 'クーポン'
        ? REMARK_COUPON
        : '';
    const newRemarkDesign = e.target.value === '0' ? REMARK_PAMPHLET : '';
    setFormData({
      ...formData,
      design: Number(e.target.value),
      remark: remarkOption + newRemarkDesign,
    });
  };
```

これを使う場合も
```
onChange={setDesign}
```
みたいにinputを受け取る部分は書かなくて大丈夫